### DISCUSSION TITLE
Clone Genmon and OS

### DISCUSSION THREAD

**Discussion by Davidw4687** [jgyates/genmon#1194] [Questions] [answered]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).
- [X] I have included the generator make and model in my submission.

### Version

1.19.04

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have submitted logs

### What is/are your question(s)?

I have Genmon running on 4 different Generac generators. I am trying to setup a 5th Genmon/generac generator monitor. I have colned the SD card with WIN32DISKIMAGER. All PI models are 4B 1meg. When I start the new Genmon system I can't communicate with the generator - RS232 serial on board. For the test I am connecting to the same generator as was being used by a working copy. All cables are the same ones used with the Genmon system that is running fine - no problems for 4 years.  Am I missing something? Generac gen is a 24kw new this year.

**Comment by jgyates** [comment]
This is just a guess. the firmware versions may be different between the two pi. I only backup the contents of /etc/genmon and just install a fresh copy of the OS as I always want the latest version of Linux as it contains bug fixes and security updates.

**Comment by skipfire** [comment]
@Davidw4687 have you tried the serial loopback test to confirm that the Pi & RS232 are working?

Next I'd suggest taking a working unit with the non-working unit and swapping parts back and forth to identify for sure what the problem is.
1. Put the non-working micro-SD in a working existing setup (just the micro-SD card, not the Pi or RS232 parts) - if this works then it's probably the new hardware, if it doesn't work it's probably the micro-SD image.
2. Put the working micro-SD card into the non-working hardware and see if that works - if it does then the hardware is good and it confirms it's the micro-SD image, if it doesn't work then it's probably the hardware.
3. If both tests fail then something funky's going on.

**Comment by Davidw4687** [comment, marked as answer]
Thanks for getting back to me. First, thanks. Second, I’m an idiot. After rereading the documentation and the serial communication trouble shooting guide I figured out the problem. The new PI/Genmon was setup on my bench with an external power supply. I did not carry the ground from the generator controller to the RS232 interface. Once the ground was made the system works fine.
